tools/python/boutiques/validator.py

```diff
@@ -122,6 +122,19 @@ def inById(i):
             else:
                 errors += []
 
+    if 'output-files' in descriptor:
+        # Verify output file with non-optional conditional file template
+        # contains a default path
+        msg_template = ("OutputError: \"{0}\": Non-optional output-file with "
+                        "conditional-path-template must contain "
+                        "\"default\" path-template.")
+        for outF in [o for o in descriptor["output-files"] if
```

Comment:
Could we add logic here verifying that the conditions being evaluated are valid? I.e. the `conditional-path-template` fields are either "default" or contain a value that corresponds to one of the input IDs?

Also, another thing to check, is that the "types" are compatible among comparisons. I.e.:

```python
number < 6
```
is not equivalent to 
```python
number < "6"
```
and which one is desired is entirely based on the datatype of `number.`

We may want to at least throw a warning if the type of the value being compared is not the same type as the input used in the comparison? The easiest way to do this may be to generate a random value of that type and perform the comparison, and then catch + return the error we encounter (if we encounter one)?